sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/HDFSMetadataLog.scala

```diff
@@ -49,10 +49,11 @@ import org.apache.spark.sql.internal.SQLConf
 class HDFSMetadataLog[T <: AnyRef : ClassTag](sparkSession: SparkSession, path: String)
   extends MetadataLog[T] with Logging {
 
-  private implicit val formats = Serialization.formats(NoTypeHints)
+  private implicit val formats: Formats = Serialization.formats(NoTypeHints)
 
   /** Needed to serialize type T into JSON when using Jackson */
-  private implicit val manifest = Manifest.classType[T](implicitly[ClassTag[T]].runtimeClass)
+  private implicit val manifest: Manifest[T] =
```

Comment:
This may need rewrite as 

```scala
  private implicit val manifest: Manifest[T] = {
    val classTag = implicitly[ClassTag[T]]
    if (classTag == null) {
      Manifest.classType[T](classOf[AnyRef])
    } else {
      Manifest.classType[T](classTag.runtimeClass)
    }
  }
```

Reply:
it maybe cause this error?

class [Ljava.lang.Object; cannot be cast to class [Lorg.apache.spark.sql.execution.streaming.FileStreamSource$FileEntry; ([Ljava.lang.Object;

Reply:
I'm not sure, I didn't run all the tests

Reply:
refferences to FileMetadataStructSuite:
I think this is the generation of generic loss. I am learning scala reflection and related code these days。
org.apache.spark.sql.execution.streaming.CompactibleFileStreamLog#allFiles
the method will return Array.empty,it loss T.